lesson1/contracts/Token.sol

```diff
@@ -0,0 +1,102 @@
+//SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.0;
+
+//import "hardhat/console.sol";
+
+contract Token {
+    event Transfer(address indexed _from, address indexed _to, uint256 _value);
+    event Approval(
+        address indexed _owner,
+        address indexed _spender,
+        uint256 _value
+    );
+
+    struct ApprovalStruct {
+        address owner;
+        address approved_address;
+        uint256 approved_amount;
+    }
```

Comment:
Po co aż tyle informacji przetrzymywać? Pamiętasz, że pamiętanie każdego byte'a kosztuje majątek? :)

Reply:
Trzymasz tylko uint256 dla approval czy wartości konta. To naprawdę zmniejsza drastycznie ilość danych, którą tu masz.